### PR TITLE
Cherry pick of #59686 #67224: Add cloud-provider policies to be applied via addon mgr

### PR DESCRIPTION
**Pull request by grayluck**
Cherry pick of #59686 and #67224 on release-1.9.

#59686: Add cloud-provider policies to be applied via addon mgr
#67224 Add namespace for (cluster)role(binding) cloud-provider. 

```release-note
Grant service account permissions to create/update events.
```

### CODE REVIEW

**Comment by grayluck** [comment]
3. The ILB service can report firewall events after the fix.
No conflict observed between policies created by kube-apiserver when bootstrap and those created with addon when create-lb-controller.
The PR is good to go.
/unhold
/assign @tallclair @liggitt

**Comment by tallclair** [comment]
Why does this need to be cherrypicked? What is broken right now?

**Review by tallclair — COMMENTED** [review]
This is dangerous to cherrypick. If a user already has a (cluster)role(binding) named cloud-provider, it will get clobbered by these roles, likely breaking workloads that used them.

It would be a little safer to use the "EnsureExists" addonmanager mode, which wouldn't clobber existing versions. But there are still a few edge cases that might be at risk.

I missed this when I reviewed the original, but the resource names really should have been prefixed with either `system:` or `gce:`. That would also make the names more specific, and less likely to conflict with existing resources.

File: cluster/gce/addons/loadbalancing/cloud-provider-role.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  labels:
+    addonmanager.kubernetes.io/mode: Reconcile
+  name: cloud-provider
```
[the remark this answers is not in the document]
Comment (liggitt) [reply cluster/gce/addons/loadbalancing/cloud-provider-role.yaml]: big +1 for namespacing add-on objects as much as possible

**Comment by grayluck** [comment]
/hold
I am testing the namespaced objects.

**Comment by grayluck** [comment]
Manually tested by applying yamls to a GKE 1.9.7 cluster. Seeing events as expected.

```
  Normal  LoadBalancerManualChange  8s    gce-cloudprovider   Firewall change required by network admin: `gcloud compute firewall-rules create a91887774977f11e8a8ba42010a80009 --network shared-net --description "{\"kubernetes.io/service-name\":\"default/servehostname\", \"kubernetes.io/service-ip\":\"10.0.4.3\"}" --allow tcp:80 --source-ranges 0.0.0.0/0 --target-tags gke-k9-bedc8596-node --project yankaiz-xpn-host`
  Normal  LoadBalancerManualChange  8s    gce-cloudprovider   Firewall change required by network admin: `gcloud compute firewall-rules create k8s-38a98e31ecc5d522-node-hc --network shared-net --description "" --allow tcp:10256 --source-ranges 130.211.0.0/22,35.191.0.0/16,209.85.152.0/22,209.85.204.0/22 --target-tags gke-k9-bedc8596-node --project yankaiz-xpn-host`
```

Still doing sandbox testing.

**Comment by grayluck** [comment]
/hold cancel
Tested by creating e2e cluster. See correct (cluster)role(binding).

**Comment by grayluck** [comment]
@tallclair PTAL, thanks!

**Comment by tallclair** [comment]
Changes look good, but this is no longer a cherry-pick. I suggest upstreaming the changes, and then cherrypicking them along with the original.

**Comment by grayluck** [comment]
Thanks Tim. PR#67244 created. Description modified so that this is no longer auto cherrypick, but cherrypick for #59686 #67224.

/hold
Waiting for #67224 to be merged first.

**Comment by grayluck** [comment]
/assign @mbohlool 
For milestone 1.9 and cherrypick approval. Thanks!

**Comment by mbohlool** [comment]
@grayluck Please be aware that the release note on this PR will be ignored and the release tool will use two release notes on the original PRs. If you need to change the text of release notes, please do so on those PRs.

**Comment by grayluck** [comment]
That's fine. The release notes on the original PRs explains themselves. Thanks Mehdy!
\retest

**Pull request merged**